### PR TITLE
PWX-38585: Removes extra pod name label

### PR DESCRIPTION
**Pull request by Pure-AdamuKaapan**
*Master CP of #2463*

### CODE REVIEW

File: csi/controller.go [hunk 1]
```diff
@@ -614,14 +613,17 @@ func (s *OsdCsiServer) CreateVolume(
 		}
 		newVolumeId = createResp.VolumeId
 	} else {
-		clonedMetadata := getClonedPVCMetadata(locator)
+		labels := locator.GetVolumeLabels()
 		if spec.GetFADAPodName() != "" {
-			clonedMetadata[sdk.FADAPodLabelKey] = spec.GetFADAPodName()
+			labels[api.SpecPurePodName] = spec.GetFADAPodName()
 		}
+		delete(labels, intreePvcNameKey)
```
Comment (adityadani) [inline csi/controller.go:621]: Why do we need to delete these?

**Review by pp511 — APPROVED** [review]
LGTM

File: csi/controller.go (hunk 1 above)
Comment (Pure-AdamuKaapan) [reply csi/controller.go:621]: My thought was that the PVC name and namespace would no longer be applicable on the clone (and in unit tests when I added this, the labels were empty). However, let me sanity check what happens when I remove this and try a clone on a PVC volume, I'll see if it gets the label of the source or an empty string.

Comment (Pure-AdamuKaapan) [reply csi/controller.go:621]: Actually this doesn't seem necessary - while I'll have to update unit tests, the end functionality is fine without these delete statements.

Comment (adityadani) [reply csi/controller.go:621]: I thought these would be the name and namespace of the PVC (clone) that is getting created and not of the parent volume. Isn't that the case?

**Pull request merged**